### PR TITLE
refactor(tier4_planning_rviz_plugin): apply clang-tidy for mission_ch…

### PR DESCRIPTION
**Pull request by h-ohta**
…eckpoint

## Description

- apply clang-tidy for mission_checkpoint

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/tier4_planning_rviz_plugin/include/mission_checkpoint/mission_checkpoint.hpp [hunk 1]
```diff
@@ -67,21 +67,19 @@ class MissionCheckpointTool : public rviz_default_plugins::tools::PoseTool
...
 
 protected:
-  virtual void onPoseSet(double x, double y, double theta);
+  void onPoseSet(double x, double y, double theta) override;
 
 private Q_SLOTS:
   void updateTopic();
 
-private:
+private:  // NOLINT for Qt
```
Comment (kenji-miyake) [inline common/tier4_planning_rviz_plugin/include/mission_checkpoint/mission_checkpoint.hpp:78]: Where is Qt?

Comment (h-ohta) [reply common/tier4_planning_rviz_plugin/include/mission_checkpoint/mission_checkpoint.hpp:78]: Actually  I want to write with `private Q_SLOTS`, but clang-tidy still generates warning.

Comment (kenji-miyake) [reply common/tier4_planning_rviz_plugin/include/mission_checkpoint/mission_checkpoint.hpp:78]: Hmm? What is the warning?

Comment (h-ohta) [reply common/tier4_planning_rviz_plugin/include/mission_checkpoint/mission_checkpoint.hpp:78]: Now I have to write with `private:`

Comment (h-ohta) [reply common/tier4_planning_rviz_plugin/include/mission_checkpoint/mission_checkpoint.hpp:78]: The `readability-redundant-access-specifiers` occurs

Comment (kenji-miyake) [reply common/tier4_planning_rviz_plugin/include/mission_checkpoint/mission_checkpoint.hpp:78]: Ah, I see. It's for this block.
image

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**